### DISCUSSION TITLE
Auto formatting useRef with generics oddly in typescript

### DISCUSSION THREAD

**Discussion by rykugur** [LazyVim/LazyVim#1591] [Q&A] [answered]
[Link to my lazyvim configs](https://github.com/rykugur/dotfiles/tree/master/configs/nvim/lazyvim)

Hi there. I'm relatively new to neovim and lazyvim. I'm not sure which plugin this issue might be coming from. I assume prettier, but have seen a post or two on SO indicating it might be eslint?

I'm working on some tooling for my team and decided my fancy new lazyvim setup would be my IDE of choice (force myself to learn more). I have format on save enabled, and when I save, this line:

`const derpRef = useRef<number>(-1);`

gets changed to this:

` const derpRef = useRef < number > -1;`

This doesn't happen if I remove the type, and it also doesn't happen in VS Code. Can anybody help me determine which plugin and/or rule/option is causing this?

**Comment by skoch13** [comment]
It might seem that your project doesn't have https://github.com/typescript-eslint/typescript-eslint installed.

**Reply by rykugur** [reply, marked as answer]
Thanks @skoch13  - sorry for the delay. Github decided I didn't need to be notified about this.

Turns out this was due to a conflict in the prettier settings I had added and those that came with the pre-installed via boostrap (tauri, in this case). I deleted the prettierrc file I had added and all is well.

Thanks!
